### PR TITLE
Cleanup factions code

### PR DESCRIPTION
**Pull request by metalgearsloth**
- Use prototypes and not hardcoded enum
- Use the common component lookup code
- Genericised it into faction system for any other stuff that cares (left it in the NPC folder coz I ain't contributing to the flat folder structure nightmare).

Requires https://github.com/space-wizards/RobustToolbox/pull/3065

### CODE REVIEW

File: Content.Server/NPC/Systems/FactionSystem.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        }
+
+        /// <summary>
+        /// Adds this entity to the particular faction.
+        /// </summary>
+        public void AddFaction(EntityUid uid, string faction, bool dirty = true)
+        {
+            if (!_protoManager.HasIndex<FactionPrototype>(faction))
+            {
+                _sawmill.Error($"Unable to find action {faction}");
```
Comment (mirrorcult) [inline Content.Server/NPC/Systems/FactionSystem.cs:63]:
```suggestion
                _sawmill.Error($"Unable to find faction {faction}");
```

File: Content.Server/NPC/Systems/FactionSystem.cs [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        }
+
+        /// <summary>
+        /// Removes this entity from the particular faction.
+        /// </summary>
+        public void RemoveFaction(EntityUid uid, string faction, bool dirty = true)
+        {
+            if (!_protoManager.HasIndex<FactionPrototype>(faction))
+            {
+                _sawmill.Error($"Unable to find action {faction}");
```
Comment (mirrorcult) [inline Content.Server/NPC/Systems/FactionSystem.cs:84]:
```suggestion
                _sawmill.Error($"Unable to find faction {faction}");
```

File: Content.Server/NPC/Systems/FactionSystem.cs [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+                yield return comp.Owner;
+            }
+        }
+
+        /// <summary>
+        /// Makes the source faction friendly to the target faction, 1-way.
+        /// </summary>
+        public void MakeFriendly(string source, string target)
+        {
+            if (!_protoManager.TryIndex<FactionPrototype>(source, out var sourceFaction))
```
Comment (mirrorcult) [inline Content.Server/NPC/Systems/FactionSystem.cs:140]: modifying prototypes at runtime?

Comment (metalgearsloth) [reply Content.Server/NPC/Systems/FactionSystem.cs:140]: good point, I'll make an intermediary

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**